demisto_sdk/commands/test_content/IntegrationsLock.py

```diff
@@ -209,24 +216,24 @@ def create_lock_files(
                 f"integration {integration} locked"
             )
             locked_integrations.append(integration)
-        except PreconditionFailed:
+        except (PreconditionFailed, InvalidResponse) as exception:
             # if this exception occurs it means that another build has locked this integration
             # before this build managed to do it.
             # we need to unlock all the integrations we have already locked and try again later
-            test_playbook.build_context.logging_module.warning(
-                f"Could not lock integration {integration}, Create file with precondition failed."
-                f"delaying test execution."
-            )
-            unlock_integrations(locked_integrations, test_playbook, storage_client)
-            return False
-        except InvalidResponse as exc:
-            if exc.__cause__ and type(exc.__cause__) == PreconditionFailed:
+
+            if isinstance(exception, PreconditionFailed) or (
+                exception.__cause__
+                and isinstance(exception.__cause__, PreconditionFailed)
+            ):
                 test_playbook.build_context.logging_module.warning(
-                    f"The lock file of the integration {integration}, already created, delaying test execution."
+                    f"Could not lock integration {integration}, Create file with precondition failed."
+                    f"delaying test execution."
```

Comment:
where does the delay happen? I only see a call to `unlock_integration` (or is this what causes the delay?)

Reply:
this logs is inside `lock_integrations` method that returns True if all the test's integrations were successfully locked, else False, the line after the log is `return False` means we not able to lock the integration file this iteration.